CHANGELOG.md

```diff
@@ -48,7 +49,10 @@ BUG FIXES:
 - `base64gunzip` now doesn't expose sensitive values if it fails during the base64 decoding. ([#2503](https://github.com/opentofu/opentofu/pull/2503))
 - Fix loading only the necessary encryption key providers and methods for better `terraform_remote_state` support. ([2551](https://github.com/opentofu/opentofu/issues/2551))
 - Better error messages when using `null` in invalid positions in the argument to the `transpose` function. ([#2553](https://github.com/opentofu/opentofu/pull/2553))
+- S3 based state locking now works also with 3rd party providers that support SHA256 checksums. ([#2606](https://github.com/opentofu/opentofu/pull/2606))
```

Comment:
We introduced a new feature (line #19) and fixed a bug within it before the actual release, so I believe we shouldn't include that bug fix message (it didn't actually exist for end users).

Additional note: 3rd party s3 compatible storage implementations must also support conditional writes as well as SHA256 checksums.

Reply:
Good point. 
About your note, that's coming from the actual implementation. Nonetheless, I just removed this. I was having second thoughts about adding this or not too. Thanks for helping with the decision here.